tests/features/assets.feature

```diff
@@ -844,3 +844,117 @@ Feature: Assets
+                "name": "Jpeg Example",
+                "description": "Jpeg file asset example"
+            },
+            "filename": "file_example-jpg.jpg",
+            "external_user_id": "test_user"
+        }
+        """
+        Then we get existing resource
+        """
+        {
+            "set_id": "#SETS._id#",
+            "filename": "file_example-jpg.jpg",
+            "name": "Jpeg Example",
+            "description": "Jpeg file asset example",
+            "firstcreated": "#DATE#",
+            "versioncreated": "#DATE#",
+            "original_creator": "test_user",
+            "version_creator": "test_user",
+            "state": "draft",
+            "binary": null,
+            "_media_id": "#ASSETS._media_id#",
+            "length": 12186,
+            "mimetype": "image/jpeg",
+            "_links": {
+                "self": {
+                    "title": "Asset",
+                    "href": "consume/assets/#ASSETS._id#"
+                }
+            }
+        }
+        """
+        When we lock/unlock an asset with client.assets.lock_asset
+        """
+        {
+            "docs": {
+                "lock_action": "edit"
+            },
+            "item_id": "#ASSETS._id#",
+            "external_user_id": "test_user",
+            "external_session_id": "test_1234"
+        }
+        """
+        Then we get existing resource
+        """
+        {
+            "lock_action": "edit",
+            "lock_user": "test_user",
+            "lock_session": "test_1234",
+            "lock_time": "#DATE#",
+            "versioncreated": "#DATE#",
+            "version_creator": "test_user"
+        }
+        """
+        When we lock/unlock an asset with client.assets.lock_asset
+        """
+        {
+            "docs": {
+                "lock_action": "edit"
+            },
+            "item_id": "#ASSETS._id#",
+            "external_user_id": "test_user",
+            "external_session_id": "test_1234"
+        }
+        """
+        Then we get error 400
+        When we lock/unlock an asset with client.assets.unlock_asset
+        """
+        {
+            "docs": {
+                "lock_action": "edit"
+            },
+            "item_id": "#ASSETS._id#",
+            "external_user_id": "test_user",
+            "external_session_id": "test_1234"
+        }
+        """
+        Then we get existing resource
+        """
+        {
+            "lock_action": "None",
+            "lock_user": "None",
+            "lock_session": "None",
+            "lock_time": "None"
+        }
+        """
+        When we lock/unlock an asset with client.assets.unlock_asset
+        """
+        {
+            "docs": {
+                "lock_action": "edit"
+            },
+            "item_id": "#ASSETS._id#",
+            "external_user_id": "test_user",
+            "external_session_id": "test_1234"
+        }
+        """
+        Then we get error 400
```

Comment:
It would be ideal if we also had tests for the possible exceptions that can be raised